pandas/core/indexes/datetimes.py

```diff
@@ -1776,7 +1776,8 @@ def insert(self, loc, item):
 
         if isinstance(item, (datetime, np.datetime64)):
             self._assert_can_do_op(item)
-            if not self._has_same_tz(item):
+            if not self._has_same_tz(item) and item is not self._na_value:
+                # GH#16537 allow pd.NaT through
```

Comment:
comment not needed

Reply:
Will change, but sure we don't want the issue reference?